tools/sctool/src/main/java/com/horizen/CommandProcessor.java

```diff
@@ -204,12 +308,17 @@ private void processGenesisInfo(JsonNode json) {
             SidechainProofsCompanion sidechainProofsCompanion = new SidechainProofsCompanion(new HashMap<>());
             SidechainTransactionsCompanion sidechainTransactionsCompanion = new SidechainTransactionsCompanion(new HashMap<>(), sidechainBoxesDataCompanion, sidechainProofsCompanion);
 
-            // TODO: hardcoded
-            ForgerBox forgerBox = SidechainCreation.getHardcodedGenesisForgerBox();
-            PrivateKey25519 forgerSecret = SidechainCreation.genesisSecret;
+            //Find Sidechain creation information
+            SidechainCreation sidechainCreation = null;
+            for (SidechainRelatedMainchainOutput output : mcRef.data().sidechainRelatedAggregatedTransaction().get().mc2scTransactionsOutputs()) {
+                if (output instanceof SidechainCreation) {
+                    sidechainCreation =  (SidechainCreation) output;
+                }
+            }
+
```

Comment:
Shall we throw correct error if there is no SidechainCreation transaction or we have more than one SidechainTransaction from MCOutput?

Reply:
Check for `null` with proper error message is good to have.

We can not care about multiple instances, if we trust MC consensus, that forbids to keep more than one SC creation output with the same sc id.